etc/docker/Dockerfile.triton-jetpack-focal

```diff
@@ -1,154 +1,147 @@
-
-# Needed for Viam SDK
-RUN apt-get install -y --no-install-recommends \
-    libboost-log1.71.0
-
-# We like this for memory allocation for our program
-RUN apt-get install -y --no-install-recommends \
+    python3-pip \
+    \
+    libboost-log1.71.0 \
+    \
     libjemalloc2
 
 
-FROM base AS build
-
+# Make a build-deps layer so we can get this locked in even
+# if there are problems building the module itself. Some of the tasks here
+# like building grpc, the C++ SDK, and downloading the triton runtime
+# are expensive, and it is unfortunate to need to re-execute them if
+# just the Viam triton server build itself fails.
+FROM base AS build-deps
 ARG DEBIAN_FRONTEND=noninteractive
 ENV HOME /root
 
-# Configure a basic build environment
-RUN apt-get -y --no-install-recommends install \
-    build-essential \
-    g++ \
-    git \
-    ninja-build \
-    pkg-config
+# Create directories we will use
+RUN mkdir -p ${HOME}/opt/src /opt/tritonserver
 
 # Add public key and repository to get cmake 3.25+
 RUN apt-key adv --fetch-key https://apt.kitware.com/keys/kitware-archive-latest.asc
 RUN echo 'deb https://apt.kitware.com/ubuntu/ focal main' > /etc/apt/sources.list.d/kitware.list
-RUN apt-get -y update
-RUN apt-get -y --no-install-recommends install \
-    cmake
-
-# Build grpc (and proto, etc.) from source, since we don't have a good version on focal
 
-RUN apt-get -y --no-install-recommends install \
+# Install packages for:
+# - A basic build environment
+# - Prereqs for building grpc/proto, etc
+# - Prereqs for building the Viam C++ SDK
+# - Prereqs for building the module against Triton Server
+#
+RUN apt-get -y update && apt-get -y --no-install-recommends install \
+    build-essential \
+    cmake \
+    g++ \
+    git \
+    ninja-build \
+    pkg-config \
+    \
     libc-ares-dev \
     libre2-dev \
     libssl-dev \
-    zlib1g-dev
+    zlib1g-dev \
+    \
+    libboost-all-dev \
+    \
+    curl \
+    nvidia-cuda-dev \
+    rapidjson-dev
 
-RUN mkdir -p ${HOME}/opt/src
+# Build grpc (and proto, etc.) from source, since we don't have a good version on focal
 RUN cd ${HOME}/opt/src && \
     git clone --recurse-submodules -b v1.52.0 --depth 1 --shallow-submodules https://github.com/grpc/grpc && \
     cd grpc && \
-    mkdir -p build && \
-    cd build && \
-    cmake .. -G Ninja \
+    cmake -S . -B build -G Ninja \
         -DgRPC_CARES_PROVIDER=package \
         -DgRPC_RE2_PROVIDER=package \
         -DgRPC_SSL_PROVIDER=package \
         -DgRPC_ZLIB_PROVIDER=package \
         -DgRPC_INSTALL=ON \
         -DgRPC_BUILD_TESTS=OFF \
         -DBUILD_SHARED_LIBS=ON \
-        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-        -DCMAKE_INSTALL_PREFIX=/opt/viam \
-        -DCMAKE_INSTALL_RPATH=/opt/viam/lib && \
-    ninja -j3 install
+        -DCMAKE_INSTALL_RPATH=/opt/viam/lib \
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo && \
+    cmake --build build --target install -j3 -- -v && \
+    cmake --install build --prefix /opt/viam
```

Comment:
This is a slightly more CMake-y way of runnign the build rather than calling Ninja directly.